test/functional/ui/live_substitution_spec.lua

```diff
@@ -0,0 +1,511 @@
+-- Test the good behavior of the live action : substitution
+
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear, feed, insert = helpers.clear, helpers.feed, helpers.insert
+local execute, request, neq = helpers.execute, helpers.request, helpers.neq
+
+
+describe('Live Substitution', function()
+    local screen
+    local curbuf
+
+    local hl_colors = {
```

Comment:
If you don't mind please remove this indirection and just use `Screen.colors` directly. This maybe used to make sense in the pre-`snapshot_util()`-days when one needed to retrieve and specify the  `:hi` attributes manually, but now this is just pointless extra work compared just using the attr sets printed by `snapshot_util()`. (I will do a cleanup refactor soon to remove it from the existing screen tests).